enlighten/Controller.py

```diff
@@ -1886,7 +1899,8 @@ def process_reading(self, reading, spec=None, dark=None, ref=None, settings=None
 
         # This should be done before any processing that involves multiple
         # pixels, e.g. offset, boxcar, baseline correction, or Richardson-Lucy.
-        self.vignette_roi.process(pr, settings)
+        if self.roi_enabled:
+            self.vignette_roi.process(pr, settings)
```

Comment:
I haven't tried to track down where the issue is, but this isn't sufficient.  The attached screenshot shows the CSV files generated if I save a spectrum (with ROI on), then click the button to turn the ROI off, then save a second spectrum, and then export both together.  The three files generated show that the "vignetted" metadata does toggle from TRUE to FALSE between the two spectra, but all three files are saved starting at pixel 17 (ROI Pixel Start).
![image](https://user-images.githubusercontent.com/2125705/193139891-f93952fa-d6d3-455a-beb8-00e87c0a0b64.png)